src/uart/mod.rs

```diff
@@ -0,0 +1,81 @@
+//! Using the UART
+use crate::error::{NegativeErrorExt, NumericError};
+use embedded_io;
+use embedded_io::ErrorKind;
+use riot_sys::libc::c_void;
+use riot_sys::uart_init;
+use riot_sys::uart_t;
+use riot_sys::uart_write;
+
+#[derive(Debug)]
+pub struct UARTDevice {
+    dev: uart_t,
+}
+
+#[derive(Debug)]
+pub struct Error(NumericError);
+
+impl From<NumericError> for Error {
+    fn from(err: NumericError) -> Self {
+        Self(err)
+    }
+}
+
+impl embedded_io::Error for Error {
+    fn kind(&self) -> ErrorKind {
+        match -self.0.number() as _ {
+            riot_sys::ENODEV => ErrorKind::NotFound,
+            riot_sys::ENOTSUP => ErrorKind::Unsupported,
+            _ => ErrorKind::Other,
+        }
+    }
+}
+
+impl embedded_io::ErrorType for UARTDevice {
+    type Error = Error;
```

Comment:
```suggestion
    type Error = NumericError;
```

If this is just a newtype so that embedded_io::Error can be implemented for it: We could also have that for NumericError right away.